src/services/offerService.js

```diff
@@ -229,11 +231,83 @@ export const newOffer = async ({
         if (!res.ok) {
             throw json.errors;
         }
-        // TODO count metrics
+
+        createEvent(EVENT_TYPES.SUCCESS(OFFER_NEW_METRIC_ID));
         return json;
 
     } catch (error) {
-        // TODO count metrics
+        createEvent(EVENT_TYPES.ERROR(
+            OFFER_NEW_METRIC_ID,
+            ErrorTypes.NETWORK_FAILURE
+        ));
+
+        if (Array.isArray(error)) throw error;
+        throw [{ msg: "Unexpected Error. Please try again later." }];
+    }
+};
+
+export const editOffer = async ({
+    offerId,
+    title,
+    publishDate,
+    publishEndDate,
+    jobMinDuration,
+    jobMaxDuration,
+    jobStartDate,
+    description,
+    contacts,
+    isPaid,
+    vacancies,
+    jobType,
+    fields,
+    technologies,
+    location,
+    coordinates,
+    requirements,
+}) => {
+    const data = {
+        title,
+        publishDate,
+        publishEndDate,
+        jobMinDuration,
+        jobMaxDuration,
+        jobStartDate,
+        description,
+        contacts,
+        isPaid,
+        vacancies,
+        jobType,
+        fields,
+        technologies,
+        location,
+        coordinates,
+        requirements,
+    };
+
+    try {
+        const res = await fetch(`${API_HOSTNAME}/offers/edit/${offerId}`, {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            credentials: "include",
+            body: JSON.stringify(data),
+        });
+        const json = await res.json();
+
+        if (!res.ok) {
+            throw json.errors;
+        }
+
+        createEvent(EVENT_TYPES.SUCCESS(OFFER_EDIT_METRIC_ID));
+        return json;
+
+    } catch (error) {
+        createEvent(EVENT_TYPES.ERROR(
+            OFFER_EDIT_METRIC_ID,
+            ErrorTypes.NETWORK_FAILURE
+        ));
+
```

Comment:
Here, you should the same as in the `create offer` service

Reply:
I think its done now